packages/firestore/ios/Plugin/FirebaseFirestore.swift

```diff
@@ -278,26 +302,19 @@ import FirebaseFirestore
                         completion(result, nil)
                     }
                 }
-                self.listenerRegistrationMap[callbackId] = listenerRegistration
+                await listenerRegistrationMap.addRegistration(listenerRegistration, listenerId: callbackId)
             } catch {
                 completion(nil, error)
             }
         }
     }
 
-    @objc public func removeSnapshotListener(_ options: RemoveSnapshotListenerOptions) {
+    @objc public func removeSnapshotListener(_ options: RemoveSnapshotListenerOptions) async {
```

Comment:
```suggestion
    @objc public func removeSnapshotListener(_ options: RemoveSnapshotListenerOptions, completion: @escaping () -> Void) {
```

Please use a completion handler instead.